lib/data/repository/repository_impl.dart

```diff
@@ -1,75 +1,56 @@
-import 'package:dio/dio.dart';
-import 'package:marvel/data/sources/marvel_api_client.dart';
+import 'package:hive_flutter/hive_flutter.dart';
+import 'package:marvel/data/data_sources/local_data_source.dart';
+import 'package:marvel/data/data_sources/remote_data_source.dart';
+import 'package:marvel/data/models/local_models/local_character.dart';
 import 'package:marvel/domain/entities/character.dart';
 import 'package:marvel/domain/entities/series.dart';
 import 'package:marvel/domain/repository/marvel_repository.dart';
-import 'package:marvel/data/convertors/convertor.dart';
 
 class RepositoryImpl implements MarvelRepository {
-  static final RepositoryImpl _repository = RepositoryImpl._();
-  final MarvelApiClient marvelClient = MarvelApiClient(Dio(), baseUrl: 'https://gateway.marvel.com:443/v1/public/');
-  final String ts = '2';
-  final String apiKey = 'c1bba7288e4f2f4f744591622a48412b';
-  final String hash = 'bab03858fdeab2fe461725bad8d65904';
+  final RemoteDataSourceImpl? remoteDataSource;
+  final LocalDataSourceImpl? localDataSource;
 
-  RepositoryImpl._();
-
-  factory RepositoryImpl() => _repository;
+  RepositoryImpl({this.localDataSource, this.remoteDataSource});
 
   @override
   Future<List<Character>> getCharacters(int offset) async {
     try {
-      final httpResponse = await marvelClient.getCharacters(ts, apiKey, hash, offset);
-      if (httpResponse.response.statusCode != 200) {
-        return Future.error("network error");
+      List<Character> characters = await remoteDataSource!.getCharacters(offset);
+      List<LocalCharacter> list = characters
+          .map((e) => LocalCharacter(
+              id: e.id,
+              name: e.name,
+              description: e.description,
+              smallThumbnailUrl: e.smallThumbnailUrl,
+              bigThumbnailUrl: e.bigThumbnailUrl))
+          .toList();
+      if (Hive.box('characters').isEmpty) {
```

Comment:
It would be better to refresh the cache (LocalDataSource) each time you get the appropriate data from the server.

Reply:
done